source/MaterialXContrib/MaterialXMaya/vp2ShaderFragments/materialXLightDataBuilder.xml

```diff
@@ -0,0 +1,56 @@
+<fragment uiName="materialXLightDataBuilder" name="materialXLightDataBuilder" type="plumbing" class="ShadeFragment" version="1.0" feature_level="0">
+    <description><![CDATA[Accumulates Maya lights into a MaterialX compatible structure.]]></description>
+    <properties>
+        <float3 name="L" flags="varyingInputParam" />
+        <int name="lightType" />
+        <float3 name="diffuseI" />
+        <float3 name="specularI" />
+    </properties>
+    <values>
+    </values>
+    <outputs>
+        <float3 name="outColor" />
+    </outputs>
+    <implementation>
+        <implementation render="OGSRenderer" language="GLSL" lang_version="3.0">
```

Comment:
Is it worth-while to write the HLSL version of this now, or is there more work required for this?